user/db.go

```diff
@@ -73,161 +70,276 @@ func (s *Store) updateUser(ctx context.Context, user User) error {
+		`SELECT *
 		FROM people.users
-		ORDER BY user_id ASC;`)
+		WHERE (CAST(user_id AS TEXT) LIKE '%' || $1 || '%')
+		   OR (username LIKE '%' || $1 || '%')
+		   OR (nickname LIKE '%' || $1 || '%')
+		   OR (first_name LIKE '%' || $1 || '%')
+		   or (last_name LIKE '%' || $1 || '%')
+		   OR (email LIKE '%' || $1 || '%')
+		   OR (first_name || ' ' || last_name LIKE '%' || $1 || '%');`, search)
 	if err != nil {
 		return nil, err
 	}
 	return u, nil
 }
 
-// getUsersIDD will get a group of users by userId DESC
-func (s *Store) getUsersIDD(ctx context.Context) ([]User, error) {
+// getUsersSearchSizePage will get users search with size and page
+func (s *Store) getUsersSearchSizePage(ctx context.Context, search string, size, page int) ([]User, error) {
 	var u []User
 	err := s.db.SelectContext(ctx, &u,
-		`SELECT user_id, username, nickname, first_name, last_name, email, last_login, avatar, use_gravatar
+		`SELECT *
 		FROM people.users
-		ORDER BY user_id DESC;`)
+		WHERE (CAST(user_id AS TEXT) LIKE '%' || $1 || '%')
+		   OR (username LIKE '%' || $1 || '%')
+		   OR (nickname LIKE '%' || $1 || '%')
+		   OR (first_name LIKE '%' || $1 || '%')
+		   or (last_name LIKE '%' || $1 || '%')
+		   OR (email LIKE '%' || $1 || '%')
+		   OR (first_name || ' ' || last_name LIKE '%' || $1 || '%')
+-- 		GROUP BY u, user_id, username, university_username, email, first_name, last_name, nickname, login_type, password, salt, avatar, last_login, reset_pw, enabled, created_at, created_by, updated_at, updated_by, deleted_at, deleted_by, use_gravatar, ldap_username
+		LIMIT $2
+		OFFSET $3;`, search, size, size*(page-1))
 	if err != nil {
 		return nil, err
 	}
 	return u, nil
 }
 
-// getUsersFLNA will get a group of users by first and last name ASC
-func (s *Store) getUsersFLNA(ctx context.Context) ([]User, error) {
+// getUsersOptionsDesc will get users sorting asc
+func (s *Store) getUsersOptionsAsc(ctx context.Context, sortBy string) ([]User, error) {
 	var u []User
 	err := s.db.SelectContext(ctx, &u,
-		`SELECT user_id, username, nickname, first_name, last_name, email, last_login, avatar, use_gravatar
+		`SELECT *
 		FROM people.users
-		ORDER BY first_name ASC, last_name ASC;`)
+		ORDER BY
+		    CASE WHEN $1 = 'userId' THEN user_id END ASC,
+		    CASE WHEN $1 = 'name' THEN first_name END ASC,
+			CASE WHEN $1 = 'name' THEN last_name END ASC,
+		    CASE WHEN $1 = 'username' THEN username END ASC,
+		    CASE WHEN $1 = 'email' THEN email END ASC,
+		    CASE WHEN $1 = 'lastLogin' THEN last_login END ASC NULLS FIRST;`, sortBy)
 	if err != nil {
 		return nil, err
 	}
 	return u, nil
 }
 
-// getUsersIDD will get a group of users by first and last DESC
-func (s *Store) getUsersFLND(ctx context.Context) ([]User, error) {
+// getUsersOptionsDescSizePage will get users sorting asc with size and page
+func (s *Store) getUsersOptionsAscSizePage(ctx context.Context, sortBy string, size, page int) ([]User, error) {
```

Comment:
NITPICK: Could the previous functions be folded into this one, or does the SQL get too gnarly?

Reply:
SQL gets weird, in the version I made finished a few days ago sorts this to as far down as I can reasonably get it but goes through some iterations as it took a lot of time to get right